### DISCUSSION TITLE
[Feature]: Search function on web

### DISCUSSION THREAD

**Discussion by athornfam2** [immich-app/immich#1637] [Feature Request]
### Feature detail

Looking for the ability to filter photos by time and date, camera model, lens, and other parameters to instantly display results.

### Platform

Web

**Comment by veverkap** [comment]
@alextran1502 are you looking for any help on this one?

**Comment by alextran1502** [comment]
> @alextran1502 are you looking for any help on this one?

Thank you for the offer, we are working on a new tagging system which will replace the current search system. And I am planning to work on the web once we replace the current search system as well

**Reply by Pikachews** [reply]
@alextran1502 is there an issue for this feature that I can follow? Thanks 🙏

**Comment by sliterok** [comment]
Is there any chance that search will be implemented with some sophisticated logic like in this repo [rom1504/clip-retrieval](https://github.com/rom1504/clip-retrieval)?

**Comment by uhthomas** [comment]
I think this has been resolved?

**Reply by julian-alarcon** [reply]
How was it resolved @uhthomas ? I cannot filter by device/camera yet

**Reply by uhthomas** [reply]
You can search for device/camera with metadata search, or `m:<device>`.

<img width="1178" alt="Screenshot 2023-08-17 at 17 22 05" src="https://github.com/immich-app/immich/assets/9749173/f048d777-c851-457c-b56d-fcb9f7edf3a8">

**Reply by bo0tzz** [reply]
That's a loose text search in all the metadata, not robust filtering.

**Reply by uhthomas** [reply]
Hmm, okay. That's definitely a different feature than searching for metadata. Though we did start indexing camera metadata like the model and lens, so at least there's that and should help unblock people depending on what they needed.

**Reply by linguowei** [reply]
Can refer to Synology Photos  https://demo.synology.com/en-global/dsm

**Reply by linguowei** [reply]
<img width="1902" alt="截屏2023-10-09 10 36 06" src="https://github.com/immich-app/immich/assets/17171323/80b87073-46ce-4767-9d66-f4f09572d596">
